test/common/event/scaled_range_timer_manager_impl_test.cc

```diff
@@ -316,8 +316,54 @@ TEST_F(ScaledRangeTimerManagerTest, SingleTimerSameMinMax) {
   EXPECT_FALSE(timer->enabled());
 }
 
+TEST_F(ScaledRangeTimerManagerTest, MaxOnlyScaleFactorTimer) {
+  ScaledRangeTimerManagerImpl manager(dispatcher_);
+
+  // With the default scale factor of 1, the timer should fire at its maximum timeout.
+  MockFunction<TimerCb> callback;
+  auto timer = manager.createTimer(ScaleFactor(0.5), callback.AsStdFunction());
```

Comment:
What happens if ScaleFactor is set to 100?

Reply:
My read of RangeTimerImpl::enableTimer is that the min is honored, not the max.

Reply:
...and I see that the code was buggy. Added some tests and fixed it.